pkg/providers/instancetype/instancetype.go

```diff
@@ -93,13 +97,106 @@ func (p *DefaultProvider) List(ctx context.Context, kc *v1alpha1.KubeletConfigur
 }
 
 func (p *DefaultProvider) UpdateInstanceTypes(ctx context.Context) error {
+	p.muInstanceTypeInfo.Lock()
+	defer p.muInstanceTypeInfo.Unlock()
+
+	instanceTypes, err := getAllInstanceTypes(p.ecsClient)
+	if err != nil {
+		return err
+	}
+
+	if p.cm.HasChanged("instance-types", instanceTypes) {
+		// Only update instanceTypesSeqNun with the instance types have been changed
+		// This is to not create new keys with duplicate instance types option
+		atomic.AddUint64(&p.instanceTypesSeqNum, 1)
+		log.FromContext(ctx).WithValues(
+			"count", len(instanceTypes)).V(1).Info("discovered instance types")
+	}
+	p.instanceTypesInfo = instanceTypes
 
-	// TODO: implement me
 	return nil
 }
 
+// nolint: gocyclo
 func (p *DefaultProvider) UpdateInstanceTypeOfferings(ctx context.Context) error {
+	p.muInstanceTypeOfferings.Lock()
+	defer p.muInstanceTypeOfferings.Unlock()
+
+	// Get offerings from ECS
+	instanceTypeOfferings := map[string]sets.Set[string]{}
+	describeAvailableResourceRequest := &ecsclient.DescribeAvailableResourceRequest{
+		RegionId:            tea.String(p.region),
+		DestinationResource: tea.String("InstanceType"),
+	}
+	runtime := &util.RuntimeOptions{}
 
-	// TODO: implement me
+	resp, err := p.ecsClient.DescribeAvailableResourceWithOptions(describeAvailableResourceRequest, runtime)
+	if err != nil {
+		return err
+	}
+
+	if resp.Body == nil || resp.Body.AvailableZones == nil || len(resp.Body.AvailableZones.AvailableZone) == 0 {
+		return errors.New("DescribeAvailableResourceWithOptions failed to return any instance types")
+	}
+
+	for _, az := range resp.Body.AvailableZones.AvailableZone {
+		if *az.Status == "Available" { // Available or SoldOut
+			for _, ar := range az.AvailableResources.AvailableResource {
```

Comment:
Perhaps, we need to judge `AvailableResources == nil` `SupportedResources ==nil`.